README.md

# hcandersenr <img src='man/figures/logo.png' align="right" height="139" />

<!-- badges: start -->

[![Travis build
status](https://travis-ci.org/EmilHvitfeldt/hcandersenr.svg?branch=master)](https://travis-ci.org/EmilHvitfeldt/hcandersenr)
[![AppVeyor build
status](https://ci.appveyor.com/api/projects/status/github/EmilHvitfeldt/hcandersenr?branch=master&svg=true)](https://ci.appveyor.com/project/EmilHvitfeldt/hcandersenr)
[![CRAN
status](https://www.r-pkg.org/badges/version/hcandersenr)](https://cran.r-project.org/package=hcandersenr)
[![CRAN\_Download\_Badge](http://cranlogs.r-pkg.org/badges/hcandersenr)](https://CRAN.R-project.org/package=hcandersenr)
[![DOI](https://zenodo.org/badge/98581517.svg)](https://zenodo.org/badge/latestdoi/98581517)
<!-- badges: end -->

## An R Package for H.C. Andersens fairy tales

This package contains (most) the complete texts of 157 fairy tales of
H.C. Andersen, in a number of different languages. formatted to be
convenient for text analysis. Where each text is formatted to elements
of about 80 characters. The package contains:

  - `hcandersen_da`: 138 out of the 157 Fairy tales in Danish
  - `hcandersen_de`: 150 out of the 157 Fairy tales in German
  - `hcandersen_en`: 156 out of the 157 Fairy tales in English
  - `hcandersen_es`: 154 out of the 157 Fairy tales in Spanish
  - `hcandersen_fr`: 58 out of the 157 Fairy tales in French
  - `EK`: dataframe with ID, names and publishing dates
  - `hca_fairytales`: Complete collection of all fairytales in this
    package

## Avaliable fairytales

``` r
library(hcandersenr)
library(tidyverse)
hca_fairytales() %>% 
  select(book, language) %>% 
  unique() %>% 
  mutate(langauge = fct_relevel(language, c("English", "Spanish", "German", "Danish", "French"))) %>%
  ggplot(aes(langauge, book)) + 
  geom_raster(alpha = 0.3) +
  scale_x_discrete(position = "top")
```

![](man/figures/README-unnamed-chunk-2-1.png)<!-- -->

## Installation

To install the development version from Github:

``` r
library(devtools)
install_github("EmilHvitfeldt/hcandersenr")
library(hcandersenr)
```


NEWS.md
# hcandersenr 0.2.0

* First CRAN version


R/hca_fairytales.R
#' Tidy data frame of all of H.C. Andersens fairy tales in this pacakage
#'
#' A tidy data frame of all of H.C. Andersens fairy tales in this
#' pacakage with three columns: \code{text}, which contains the text of the
#' fairy tales divided into elements of up to about 80 characters each,
#' \code{book}, which contains the titles of the fairy tales in english, and
#' \code{language} which is the language of the \code{text}.
#'
#' @format A data frame with three variables: \code{text},
#' \code{book} and \code{language}.
#' @examples 
#' 
#' tales <- hca_fairytales()
#' 
#' # See first few rows
#' head(tales)
#' 
#' @export
hca_fairytales <- function() {
  length_da <- NROW(hcandersenr::hcandersen_da)
  length_de <- NROW(hcandersenr::hcandersen_de)
  length_en <- NROW(hcandersenr::hcandersen_en)
  length_es <- NROW(hcandersenr::hcandersen_es)
  length_fr <- NROW(hcandersenr::hcandersen_fr)
  
  out <- rbind(
    hcandersenr::hcandersen_da["text"],
    hcandersenr::hcandersen_de["text"],
    hcandersenr::hcandersen_en["text"],
    hcandersenr::hcandersen_es["text"],
    hcandersenr::hcandersen_fr["text"]
  )
  
  id_da <- hcandersenr::EK$name_en
  names(id_da) <- hcandersenr::EK$name_da
  id_de <- hcandersenr::EK$name_en
  names(id_de) <- hcandersenr::EK$name_de
  id_es <- hcandersenr::EK$name_en
  names(id_es) <- hcandersenr::EK$name_es
  id_fr <- hcandersenr::EK$name_en
  names(id_fr) <- hcandersenr::EK$name_fr
  
  out$book <- c(id_da[hcandersenr::hcandersen_da$book], 
                id_de[hcandersenr::hcandersen_de$book],
                hcandersenr::hcandersen_en$book,
                id_es[hcandersenr::hcandersen_es$book],
                id_fr[hcandersenr::hcandersen_fr$book])
  
  
  out$language <- rep(c("Danish", "German", "English", "Spanish", "French"),
                      times = c(length_da, length_de, length_en, length_es, length_fr))
  out
}


R/data.R
#' Fairy tales and stories of H.C. Andersen.
#'
#' This package contains the (mostly) complete text of 157 fairy tales of
#' H.C. Andersen, in a number of different languages. formatted to be
#' convenient for text analysis.
#' @docType package
#' @name hcandersenr
NULL

#' The text of H.C. Andersens fairy tales in Danish.
#'
#' A dataset containing 139 out of 157 of H.C. andersens fairy tales
#' translated in Danish. The UTF-8 plain text was sourced from
#' http://www.andersenstories.com/ and is divided into elements of up
#' to about 80 characters each.
#'
#' @format A data frame with 20439 rows and 2 variables: \code{text} and
#' \code{book}.
"hcandersen_da"

#' The text of H.C. Andersens fairy tales in German.
#'
#' A dataset containing 156 out of 157 of H.C. andersens fairy tales
#' translated to German. The UTF-8 plain text was sourced from
#' http://www.andersenstories.com/ and is divided into elements of up
#' to about 80 characters each.
#'
#' @format A data frame with 28255 rows and 2 variables: \code{text} and
#' \code{book}.
"hcandersen_de"

#' The text of H.C. Andersens fairy tales in English.
#'
#' A dataset containing 157 out of 157 of H.C. andersens fairy tales
#' translated to English. The UTF-8 plain text was sourced from
#' http://www.andersenstories.com/ and is divided into elements of up
#' to about 80 characters each.
#'
#' @format A data frame with 27859 rows and 2 variables: \code{text} and
#' \code{book}.
"hcandersen_en"

#' The text of H.C. Andersens fairy tales in Spanish.
#'
#' A dataset containing 157 out of 157 of H.C. andersens fairy tales
#' translated to Spanish. The UTF-8 plain text was sourced from
#' http://www.andersenstories.com/ and is divided into elements of up
#' to about 80 characters each.
#'
#' @format A data frame with 27510 rows and 2 variables: \code{text} and
#' \code{book}.
"hcandersen_es"

#' The text of H.C. Andersens fairy tales in French
#'
#' A dataset containing 61 out of 157 of H.C. andersens fairy tales
#' translated to French The UTF-8 plain text was sourced from
#' http://www.andersenstories.com/ and is divided into elements of up
#' to about 80 characters each.
#'
#' @format A data frame with 11184 rows and 2 variables: \code{text} and
#' \code{book}.
"hcandersen_fr"

#' Eventyrkode (fairy tale code)
#'
#' Eventyrkode (fairy tale code), stipulated by Svend Juel Møller in order
#' to identify the fairy tales by H.C. Andersen
#'
#' @format A data frame with nine variables:
#' \describe{
#' \item{\code{EK}}{Numerical value for identification of fairy tales}
#' \item{\code{date}}{Date of publication}
#' \item{\code{approximate}}{Is the \code{date} an approximation}
#' \item{\code{org_language}}{Original language of the fairy tale}
#' \item{\code{name_da}}{Name of the fairy tale in Danish}
#' \item{\code{name_de}}{Name of the fairy tale in German}
#' \item{\code{name_en}}{Name of the fairy tale in English}
#' \item{\code{name_es}}{Name of the fairy tale in Spanish}
#' \item{\code{name_fr}}{Name of the fairy tale in French}
#' }
"EK"
